keras/distribution/distribution_lib.py

```diff
@@ -426,15 +424,38 @@ def get_tensor_layout(self, path):
         return None
 
     def distribute_dataset(self, dataset):
+        import tensorflow as tf
+        from tensorflow.python.data.experimental.ops import (
+            distribute as tf_data_distribute,
+        )
+
         if not isinstance(dataset, tf.data.Dataset):
             raise ValueError(
                 "Only `tf.data.Dataset` is supported for "
                 f"sharding, got {type(dataset)}"
             )
         if self._is_multi_process:
-            return dataset.shard(
-                num_shards=self._num_process, index=self._process_id
-            ).prefetch(tf.data.AUTOTUNE)
+            batch_size = tf_data_distribute.compute_batch_size(dataset)
+            if batch_size.numpy() < 0:
+                raise ValueError(
```

Comment:
In what cases does this happen? Unbatched dataset? The error message should make explicit what user action is required (e.g. calling `.batch()`).

Reply:
yea, most likely due to the dataset is not batched. Update the error message.